manof/image.py

```diff
@@ -218,7 +220,18 @@ def run(self):
             print command
 
         try:
-            yield self._run_command(command)
+            out, _, _ = yield self._run_command(command)
+
+            if self.pipe_stdout:
+                print >> sys.stdout, out
+
+        except utils.CommandFailedError as exc:
+            if self.pipe_stderr:
+                print >> sys.stderr, exc.err
+                sys.exit(exc.code)
```

Comment:
why panic using `sys.exit`? aren't there logics wrapping `image.run` call?

Reply:
Figured I'll be true to the original return code in case user wan't stderr to behave authentically.
This is a scenario where  docker image acts like a binary and I'd like to preserve some behaviors around that.
I'm partial about the return code though. We don't care about that too much today, but someone might

Reply:
I see

We can either store the `exc.code` of last `.run` call and preserve it to its callee
or, maybe, string all exceptions handling here, and do handle it from `.run` callee

either way, I won't use the `sys.exit` here

Reply:
Unfortunately stripping error handling here is probably not a good idea, it's functional in many ways.
I'll see about preserving retcode and if it's reasonable to access it from the main callee

Reply:
@liranbg  - It's not very natural to the code as is. I decided to drop the authentic return status for now. It's too out of place you're right